cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/gvn/ValueNumbering.qll

```diff
@@ -19,9 +19,19 @@ class ValueNumberPropertyProvider extends IRPropertyProvider {
  * The value number assigned to a particular set of instructions that produce equivalent results.
  */
 class ValueNumber extends TValueNumber {
-  final string toString() { result = getExampleInstruction().getResultId() }
+  final string toString() { result = "GVN" }
 
-  final Language::Location getLocation() { result = getExampleInstruction().getLocation() }
+  final Language::Location getLocation() {
+    result =
+      min(Language::Location l |
+        l = getAnInstruction().getLocation()
+      |
+        l
+        order by
+          l.getFile().getAbsolutePath(), l.getStartLine(), l.getStartColumn(), l.getEndLine(),
+          l.getEndColumn()
+      )
```

Comment:
We should exclude `UnknownLocation` if there are any expressions with known locations in the same GVN. Otherwise implicit `this` qualifiers on member accesses will cause the value number for the `this` pointer to have an `UnknownLocation`.

Reply:
Good catch. I've excluded `UnknownLocation`s if there are any non-`UnknownLocation`s now.